pkg/daemon/ceph/osd/kms/kms.go

```diff
@@ -238,6 +238,25 @@ func (c *Config) UpdateSecret(secretName, secretValue string) error {
 
 		return nil
 	}
+	if c.IsVault() {
+		// Store the secret in Vault
+		v, err := InitVault(c.ClusterInfo.Context, c.context, c.ClusterInfo.Namespace, c.clusterSpec.Security.KeyManagementService.ConnectionDetails)
```

Comment:
can you add this cluster yam CR spec?

Reply:
No spec change is needed.  KeyRotation spec was already added [here](https://github.com/rook/rook/commit/5dc8f5a688c72fb95dc032d001b8b7ed47e78cae).